### PR TITLE
Metadata decider logic

### PR DESCRIPTION
**Pull request by cjllanwarne**
Implements HybridMetadataReadDecider logic from https://docs.google.com/document/d/1VYnzk97yTtllozO9ivZpZQTwrsY5T0wGqxlvAbrEQgg/edit#bookmark=id.mwkzc5gtae81

* Asks classic metadata service for archive status 
* Depending on the response, forwards the original query to either the classic or carbonite metadata service

### CODE REVIEW

File: hybridCarboniteMetadataService/src/main/scala/cromwell/services/metadata/hybridcarbonite/HybridReadDeciderActor.scala [hunk 1]
```diff
@@ -1,43 +1,86 @@
...
+
+        case query: QueryForWorkflowsMatchingParameters =>
+          classicMetadataServiceActor ! query
+          goto(WaitingForMetadataResponse) using WorkingData(sndr, read)
+      }
+  }
+
+  when(RequestingMetadataArchiveStatus) {
+    case Event(WorkflowQuerySuccess(response, _), wd: WorkingData) =>
+      if (response.results.size == 1 && response.results.headOption.exists(_.metadataArchiveStatus == MetadataArchiveStatus.Archived)) {
```
Comment (mcovarr) [inline hybridCarboniteMetadataService/src/main/scala/cromwell/services/metadata/hybridcarbonite/HybridReadDeciderActor.scala]: Is that size check necessary? And if not can the ```response.results.headOption.exists(_.metadataArchiveStatus == MetadataArchiveStatus.Archived))``` just be part of the match?

File: hybridCarboniteMetadataService/src/main/scala/cromwell/services/metadata/hybridcarbonite/HybridReadDeciderActor.scala [hunk 2]
```diff
@@ -1,43 +1,86 @@
...
+        carboniteMetadataServiceActor ! wd.request
+      } else {
+        classicMetadataServiceActor ! wd.request
+      }
+      goto(WaitingForMetadataResponse)
+
+    case Event(WorkflowQueryFailure(reason), wd: WorkingData) =>
+      log.error(reason, s"Failed to determine how to route ${wd.request.getClass.getSimpleName}. Falling back to classic.")
```
Comment (mcovarr) [inline hybridCarboniteMetadataService/src/main/scala/cromwell/services/metadata/hybridcarbonite/HybridReadDeciderActor.scala]: sounds Programmer Erroresque?

File: hybridCarboniteMetadataService/src/test/scala/cromwell/services/metadata/hybridcarbonite/HybridReadDeciderActorSpec.scala [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    carboniteMetadataActor.msgAvailable should be(false)
+  }
+
+  it should "go straight to the classic metadata service for archive status for multi-workflow queries" in {
+
+    val client = TestProbe("client")
+    val classicMetadataActor = TestProbe("classic")
+    val carboniteMetadataActor = TestProbe("carboniter")
+
+    val hrda: TestFSMRef[HybridReadDeciderState, HybridReadDeciderData, HybridReadDeciderActor] =
```
Comment (mcovarr) [inline hybridCarboniteMetadataService/src/test/scala/cromwell/services/metadata/hybridcarbonite/HybridReadDeciderActorSpec.scala]: not sure that ascription is helping

File: hybridCarboniteMetadataService/src/main/scala/cromwell/services/metadata/hybridcarbonite/HybridReadDeciderActor.scala [hunk 4]
```diff
@@ -1,43 +1,86 @@
...
-    case read: MetadataReadAction =>
-      sndr = Option(sender())
-      decide(read) onComplete {
-        case Success(forwardTo) =>
-          forwardTo ! read
-        case Failure(e) =>
-          respondAndStop(FailedMetadataResponse(read, new Exception("Failed to decide which metadata service to forward request to", e)))
+  when(Pending) {
+    case Event(read: MetadataReadAction, NoData) =>
+      val sndr = sender()
```
Comment (mcovarr) [inline hybridCarboniteMetadataService/src/main/scala/cromwell/services/metadata/hybridcarbonite/HybridReadDeciderActor.scala]: unnecessary

File: hybridCarboniteMetadataService/src/main/scala/cromwell/services/metadata/hybridcarbonite/HybridReadDeciderActor.scala (hunk 1 above)
Comment (cjllanwarne) [reply hybridCarboniteMetadataService/src/main/scala/cromwell/services/metadata/hybridcarbonite/HybridReadDeciderActor.scala]: 👍 I'll break this into separate a `0` case (which probably indicates "hasn't been summarized yet") and `> 1` case (which probably indicates a horrible, horrible error)

**Pull request merged**